src/Appwrite/Platform/Modules/Proxy/Http/Rules/Site/Create.php

```diff
@@ -0,0 +1,159 @@
+
+use Appwrite\Event\Certificate;
+use Appwrite\Event\Event;
+use Appwrite\Extend\Exception;
+use Appwrite\Network\Validator\CNAME;
+use Appwrite\SDK\AuthType;
+use Appwrite\SDK\Method;
+use Appwrite\SDK\Response as SDKResponse;
+use Appwrite\Utopia\Response;
+use Utopia\App;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Exception\Duplicate;
+use Utopia\Database\Helpers\ID;
+use Utopia\Database\Validator\UID;
+use Utopia\Domains\Domain;
+use Utopia\Platform\Action;
+use Utopia\Platform\Scope\HTTP;
+use Utopia\System\System;
+use Utopia\Validator\Domain as ValidatorDomain;
+use Utopia\Validator\Text;
+
+class Create extends Action
+{
+    use HTTP;
+
+    public static function getName()
+    {
+        return 'createSiteRule';
+    }
+
+    public function __construct()
+    {
+        $this
+            ->setHttpMethod(Action::HTTP_REQUEST_METHOD_POST)
+            ->setHttpPath('/v1/proxy/rules/site')
+            ->groups(['api', 'proxy'])
+            ->desc('Create site rule')
+            ->label('scope', 'rules.write')
+            ->label('event', 'rules.[ruleId].create')
+            ->label('audits.event', 'rule.create')
+            ->label('audits.resource', 'rule/{response.$id}')
+            ->label('sdk', new Method(
+                namespace: 'proxy',
+                name: 'createSiteRule',
+                description: <<<EOT
+                Create a new proxy rule for serving Appwrite Site on custom domain.
+                EOT,
+                auth: [AuthType::ADMIN],
+                responses: [
+                    new SDKResponse(
+                        code: Response::STATUS_CODE_CREATED,
+                        model: Response::MODEL_PROXY_RULE,
+                    )
+                ]
+            ))
+            ->label('abuse-limit', 10)
+            ->label('abuse-key', 'userId:{userId}, url:{url}')
+            ->label('abuse-time', 60)
+            ->param('domain', null, new ValidatorDomain(), 'Domain name.')
+            ->param('siteId', '', new UID(), 'ID of site to be executed.')
+            ->param('branch', '', new Text(255, 0), 'Name of VCS branch to deploy changes automatically', true)
+            ->inject('response')
+            ->inject('project')
+            ->inject('queueForCertificates')
+            ->inject('queueForEvents')
+            ->inject('dbForPlatform')
+            ->inject('dbForProject')
+            ->callback([$this, 'action']);
+    }
+
+    public function action(string $domain, string $siteId, string $branch, Response $response, Document $project, Certificate $queueForCertificates, Event $queueForEvents, Database $dbForPlatform, Database $dbForProject)
+    {
+        $mainDomain = System::getEnv('_APP_DOMAIN', '');
+        $sitesDomain = System::getEnv('_APP_DOMAIN_SITES', '');
+        $functionsDomain = System::getEnv('_APP_DOMAIN_FUNCTIONS', '');
+
+        $deniedDomains = [
+            $mainDomain,
+            $sitesDomain,
```

Comment:
Sites domain should be allowed here, I guess

Reply:
This is entire logic:

```php
        $deniedDomains = [
            $mainDomain,
            $sitesDomain,
            $functionsDomain,
            'localhost',
            APP_HOSTNAME_INTERNAL,
        ];

        if (\in_array($domain, $deniedDomains)) {
            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'This domain name is not allowed. Please pick another one.');
        }
```

Notice we are not denying `og.appwrite.global`, we only deny exact match, so `appwrite.global`. And that we disallwo for seucrity reasons, so developers dont steal the main domain itself